augur/export_v2.py

```diff
@@ -46,51 +46,67 @@ def convert_tree_to_json_structure(node, metadata, div=0, strains=None):
-                    values_in_tree.add(values['traits'][trait]['value'])
-
-        if trait.lower() in color_mapping:
-            # remember that the color maps (from the TSV) are in lower case, but
-            # this is not how they should be exported
-            case_map = {str(val).lower(): val for val in values_in_tree}
-            options["scale"] = {case_map[m[0]]: m[1] for m in color_mapping[trait.lower()] if m[0] in case_map}
-
-        else:
-            # TODO detect other types (ordinal, boolean)
-            if options['type'] == 'categorical' and options['title'] != 'authors':
-                #if its int or float and no colours supplied - set continuous.
-                # TODO this should maybe be expanded to long and scientific notation (?)
-                if all([ isinstance(n, float) if isinstance(n, float) else isinstance(n, int) for n in values_in_tree ]):
+def get_colorings(config, traits, provided_colors, node_metadata, mutations_present):
+    def _get_values(key):
+        values_in_tree = set()
+        for node_properties in node_metadata.values():
+            # beware of special cases!
+            if key in ["num_date", "author"]:
+                values_in_tree.add(node_properties.get(key, {}).get("value"))
+            elif key in node_properties['traits']:
+                values_in_tree.add(node_properties['traits'][key]['value'])
+        return values_in_tree
+
+    def _guess_type(key, values):
+        # TODO detect other types (ordinal, boolean)
+        if all([ isinstance(n, float) if isinstance(n, float) else isinstance(n, int) for n in values ]):
+            return "continuous"
+        return "categorical"
+
+    def _guess_title(key):
+        if key == "clade_membership":
+            return "Clade"
+        return key
+
+    # NOTE: Emma -- currently if "country" is defined as a command-line trait
+    # but there is a "color_options" in the config without it, then "country"
+    # won't be a color-by. Not sure what behavior is desired here.
+    if config.get("color_options"):
+        color_config = config["color_options"]
+    else:
+        color_config = {t: {} for t in traits}
+
+    colorings = {}
+    # handle special cases
+    if mutations_present:
+        colorings["gt"] = config.get("gt") if config.get("gt") else {'title': 'Genotype', 'type': 'ordinal'}
+    if _get_values("author"): # check if any nodes have author set
+        colorings["author"] = config.get("author") if config.get("author") else {'title': 'Authors', 'type': 'categorical'}
+    if _get_values("num_date"): # TODO: check if tree has temporal inference (possible to not have)
+        colorings['num_date'] = config.get("num_date") if config.get("num_date") else {'title': 'Sampling date', 'type': 'continuous'}
+    # remove keys which may exist but are special cases and/or have been handled above
+    color_config.pop("gt", None)
+    color_config.pop("authors", None) # this was v1 syntax
```

Comment:
We should add a check here like the other ones I have (IMO). Check for `authors` above, display a warning, and internally convert to `author`. It's inevitable there will be hangovers from v1, but let's not make things just disappear! People will never catch the 's' difference unless we point it out :)

Reply:
Done in merge 😄